src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/part/KineticPartMachine.java

```diff
@@ -62,6 +70,47 @@ public void onRotated(Direction oldFacing, Direction newFacing) {
         }
     }
 
+    @Override
+    public boolean onWorking(IWorkableMultiController controller) {
+        if (stopWorkingSub == null) {
+            stopWorkingSub = subscribeServerTick(this::stopWhenControllerNotWorking);
+        }
+        return super.onWorking(controller);
+    }
+
+    private void stopWhenControllerNotWorking() {
+        for (IMultiController controller : getControllers()) {
+            if (controller instanceof IWorkableMultiController workableMultiController) {
+                if (!workableMultiController.getRecipeLogic().isWorking()) {
+                    getKineticHolder().stopWorking();
+                    assert stopWorkingSub != null;
```

Comment:
...this will literally just crash if the subscription happens to be null, for example by the loop already succeeding.